src/solvers/dg/dg_2d.jl

```diff
@@ -46,11 +46,11 @@ function create_cache(mesh::TreeMesh{2}, nonconservative_terms::Val{true}, equat
   f2_threaded = A[A(undef, nvariables(equations), nnodes(dg), nnodes(dg), nnodes(dg))
                   for _ in 1:Threads.nthreads()]
 
-  MA2d = MArray{Tuple{nvariables(equations), nnodes(dg)}, uEltype}
-  fstar_upper_threaded               = [MA2d(undef) for _ in 1:Threads.nthreads()]
-  fstar_lower_threaded               = [MA2d(undef) for _ in 1:Threads.nthreads()]
-  noncons_diamond_upper_threaded     = [MA2d(undef) for _ in 1:Threads.nthreads()]
-  noncons_diamond_lower_threaded = [MA2d(undef) for _ in 1:Threads.nthreads()]
+  MA2d = MArray{Tuple{nvariables(equations), nnodes(dg)}, uEltype, 2, nvariables(equations) * nnodes(dg)}
```

Comment:
Why the change in the matrix size here? This seems to be unrelated to the flux plus dissipation topic?

Reply:
This change made the type concrete to increase type-stability (and performance). The size of the matrices didn't change.

Reply:
Great, now I understand! Thanks!